kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs

```diff
@@ -696,7 +696,21 @@ impl<'tcx> GotocCtx<'tcx> {
         t: Ty<'tcx>,
     ) -> Expr {
         match k {
-            PointerCast::ReifyFnPointer => self.codegen_operand(o).address_of(),
+            PointerCast::ReifyFnPointer => match self.operand_ty(o).kind() {
+                ty::FnDef(def_id, substs) => {
+                    // To be safe, run normal codegen to ensure the symbol table is updated as normally.
+                    // Maybe this isn't necessary though?
```

Comment:
Issue to track this

Reply:
I was able to remove this in the other place, so it should be safe here too.

Reply:
I've removed it in my latest commit.